osidb/models.py

```diff
@@ -1595,6 +1596,160 @@ def bzsync(self, *args, bz_api_key, **kwargs):
         self.flaw.save(*args, bz_api_key=bz_api_key, **kwargs)
 
 
+class FlawCVSSManager(ACLMixinManager, TrackingMixinManager):
+    """Flaw CVSS Manager"""
+
+    @staticmethod
+    def create_cvss(flaw, issuer, version, **extra_fields):
+        """return a new CVSS or update an existing CVSS without saving"""
+
+        try:
+            cvss = FlawCVSS.objects.get(flaw=flaw, issuer=issuer, version=version)
+
+            for attr, value in extra_fields.items():
+                setattr(cvss, attr, value)
+            return cvss
+
+        except ObjectDoesNotExist:
+            return FlawCVSS(flaw=flaw, issuer=issuer, version=version, **extra_fields)
+
+
+class AffectCVSSManager(ACLMixinManager, TrackingMixinManager):
+    """Affect CVSS Manager"""
+
+    @staticmethod
+    def create_cvss(affect, issuer, version, **extra_fields):
+        """return a new CVSS or update an existing CVSS without saving"""
+
+        try:
+            cvss = AffectCVSS.objects.get(affect=affect, issuer=issuer, version=version)
+            for attr, value in extra_fields.items():
+                setattr(cvss, attr, value)
+            return cvss
+
+        except ObjectDoesNotExist:
+            return AffectCVSS(
+                affect=affect, issuer=issuer, version=version, **extra_fields
+            )
+
+
+class CVSS(
+    ACLMixin,
+    AlertMixin,
+    TrackingMixin,
+):
+    """Model CVSS"""
+
+    class CVSSVersion(models.TextChoices):
+        VERSION2 = "V2", "version 2"
+        VERSION3 = "V3", "version 3"
+
+    class CVSSIssuer(models.TextChoices):
+        REDHAT = "RH", "Red Hat"
+        NIST = "NIST", "NIST"
+
+    CVSS_HANDLES = {CVSSVersion.VERSION2: CVSS2, CVSSVersion.VERSION3: CVSS3}
+
+    uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    vector = models.CharField(max_length=100, blank=False)
+
+    version = models.CharField(choices=CVSSVersion.choices, max_length=10)
+
+    issuer = models.CharField(choices=CVSSIssuer.choices, max_length=16)
+
+    comment = models.TextField(blank=True)
+
+    # populated by the pre_save signal
+    score = models.FloatField(default=0)
+
+    def __str__(self):
+        return f"{self.score}/{self.vector}"
+
+    @property
+    def full_version(self):
+        """Full name of the CVSS version."""
+        return f"CVSS{self.version[1:]}"
+
+    @property
+    def cvss_handle(self):
+        """
+        Corresponding CVSS handle from CVSS library based on the version.
+        """
+        return self.CVSS_HANDLES[self.version]
```

Comment:
You only use this in tests, and the instances in this model where you use it can be replaced by `cvss_object`, so I would just inline this into `cvss_object`.

Reply:
:facepalm: Ah, this feels like sending an email with the attachment mention but without the actual attachment :grin: